habitat-lab/habitat/tasks/rearrange/rearrange_sensors.py

```diff
@@ -1176,3 +1217,344 @@ def update_metric(self, *args, task, **kwargs):
+        )[[0, 2]]
+        robot_move_vel = (
+            np.linalg.norm(robot_move_vec)
+            / (1.0 / 120.0)
+            * np.sign(robot_move_vec[0])
+        )
+        # Compute the metrics for backing up and yield
+        if dis <= 1.5 and robot_move_vel < 0.0:
+            self._backup_count += 1
+        elif dis <= 1.5 and abs(robot_move_vel) < 1.0:
+            self._yield_count += 1
+
+        # Increase the step counter
+        self._step += 1
+
+        # Check if human has been found
+        found_human = False
+        if self._check_human_dis(
+            robot_pos, human_pos
+        ) and self._check_look_at_human(human_pos, robot_pos):
+            found_human = True
+            self._has_found_human = True
+            self._found_human_times += 1
+
+        # Check if the human has been found based on one condition
+        found_human_dis = False
+        if self._check_human_dis(robot_pos, human_pos):
+            found_human_dis = True
+            self._has_found_human_dis = True
+            self._found_human_times_dis += 1
+
+        # We increase the travel distance if the robot has not yet found the human
+        if not found_human and not self._has_found_human:
+            self._agent_episode_distance += np.linalg.norm(
+                self._prev_pos - robot_pos, ord=2, axis=-1
+            )
+
+        # Compute the metrics after finding the human
+        if self._has_found_human:
+            self._dis_after_found += dis
+            self._after_found_human_times += found_human
+
+        if self._has_found_human_dis:
+            self._after_found_human_times_dis += found_human_dis
+
+        # Record the step taken to find the human
+        if self._has_found_human and self._has_found_human_step == 1500:
+            self._has_found_human_step = self._step
+
+        # Record the step taken to find the human based on distance condition
+        if (
+            self._has_found_human_dis
+            and self._has_found_human_step_dis == 1500
+        ):
+            self._has_found_human_step_dis = self._step
+
+        # Compute the minimum distance only when the minimum distance has not found yet
+        if (
+            self._min_start_end_episode_step == float("inf")
+            and self._enable_shortest_path_computation
+        ):
+            robot_to_human_min_step = task.actions[
+                "agent_1_oracle_nav_randcoord_action"
+            ]._compute_robot_to_human_min_step(
+                self._robot_init_trans, human_pos, self.human_pos_list
+            )
+
+            if robot_to_human_min_step <= self._step:
+                robot_to_human_min_step = self._step
+            else:
+                robot_to_human_min_step = float("inf")
+
+            # Update the minimum SPL
+            self._min_start_end_episode_step = min(
+                self._min_start_end_episode_step, robot_to_human_min_step
+            )
+
+        # Compute the SPL before finding the human
```

Comment:
Pardon my ignorance - what is `SPL` in this context?

Reply:
Sure, no problem! 

SPL is the metric that measures the performance of how quickly the robot can find the humanoid. The higher the SPL value is, the better, and it is bounded to be in the range of 0 and 1. 

You can read the paper section 4.1 to get how exactly we compute it